library/std/src/net/ip.rs

```diff
@@ -1135,10 +1343,18 @@ impl Ipv6Addr {
     /// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0xffff, 0xc00a, 0x2ff).is_unspecified(), false);
```

Comment:
With the change to recognize ipv4 in ipv6, we could add a testcase demonstrating this:

```rust
/// also works with IPv4-mapped addresses:
/// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0xffff, 0, 0).is_unspecified(), true)
```
And similar testcases for all the other changed `Ipv6Addr` methods.